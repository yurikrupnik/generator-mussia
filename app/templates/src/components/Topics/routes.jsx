import React from 'react';
import PropTypes from 'prop-types';
// import { Redirect } from 'react-router-dom';

const Topic = ({ match }) => (
    <div>
        <h3>
            {match.params.topicId}
        </h3>
    </div>
);

Topic.propTypes = {
    match: PropTypes.shape({}).isRequired
};

const routes = [
    {
        path: '/topics',
        component: () => (
            <h3>
                Please select a topic.
            </h3>
        ),
        // component: () => (<Redirect to="/topics/components" />),
        exact: true,
        key: '/topics/topics'
    },
    {
        path: '/topics/:topicId',
        component: Topic,
        exact: true,
        key: '/topics/:topics'
    }
];

export default routes;
