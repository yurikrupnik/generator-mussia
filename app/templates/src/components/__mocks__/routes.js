/* eslint react/jsx-filename-extension: 0 */

import React from 'react';

const routes = [
    {
        path: '/hello',
        component: () => (
            <div>
                hello
            </div>
        ),
        key: 'hello'
    },
    {
        path: '/world',
        component: () => (
            <div>
                world
            </div>
        ),
        key: 'world'
    }
];

export default routes;
