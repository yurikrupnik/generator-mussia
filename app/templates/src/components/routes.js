import Loadable from './Loadable';

const TopicsLoadableComponent = Loadable({
    loader: () => import(/* webpackChunkName: "topics" */ './Topics'),
});

const LoginLoadableComponent = Loadable({
    loader: () => import(/* webpackChunkName: "login" */ '../api/auth/container'),
});

const DashboardLoadableComponent = Loadable({
    loader: () => import(/* webpackChunkName: "dashboard" */ './Dashboard'),
});

const ProjectsLoadableComponent = Loadable({
    loader: () => import(/* webpackChunkName: "projects" */ './Projects'),
});
const UsersLoadableComponent = Loadable({
    loader: () => import(/* webpackChunkName: "users" */ './Users'),
});

const AboutLoadableComponent = Loadable({
    loader: () => import(/* webpackChunkName: "about" */ './About'),
});

const RegisterLoadableComponent = Loadable({
    loader: () => import(/* webpackChunkName: "register" */ './Register'),
});

const ChatLoadableComponent = Loadable({
    loader: () => import(/* webpackChunkName: "chat" */ './ChatRoom'),
});

const routes = [
    {
        path: '/',
        component: DashboardLoadableComponent,
        exact: true,
        key: 'dashboard'
    },
    {
        path: '/register',
        component: RegisterLoadableComponent,
        key: 'register'
    },
    {
        path: '/login',
        component: LoginLoadableComponent,
        key: 'login'
    },
    {
        path: '/about',
        component: AboutLoadableComponent,
        key: 'about'
    },
    {
        path: '/topics',
        component: TopicsLoadableComponent,
        key: 'topics'
    },
    {
        path: '/projects',
        component: ProjectsLoadableComponent,
        key: 'projects'
    },
    {
        path: '/users',
        component: UsersLoadableComponent,
        key: 'users'
    },
    {
        path: '/chat',
        component: ChatLoadableComponent,
        key: 'chat'
    }
];

export default routes;
