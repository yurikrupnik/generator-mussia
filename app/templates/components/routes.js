<%_ if (filters.loadable) { _%>
import Loadable from './Loadable';

const Topics = Loadable({
    loader: () => import(/* webpackChunkName: "topics" */ './Topics'),
});

const Login = Loadable({
    loader: () => import(/* webpackChunkName: "login" */ '../api/auth/container'),
});

const Dashboard = Loadable({
    loader: () => import(/* webpackChunkName: "dashboard" */ './Dashboard'),
});

const Projects = Loadable({
    loader: () => import(/* webpackChunkName: "projects" */ './Projects'),
});

const Users = Loadable({
    loader: () => import(/* webpackChunkName: "users" */ './Users'),
});

const About = Loadable({
    loader: () => import(/* webpackChunkName: "about" */ './about'),
});

const Register = Loadable({
    loader: () => import(/* webpackChunkName: "register" */ './Register'),
});

const ChatRoom = Loadable({
    loader: () => import(/* webpackChunkName: "chat" */ './ChatRoom'),
});
<%_ } _%>
<%_ if (!filters.loadable) { _%>
import Topics from './Topics';
import Login from '../api/auth/container';
import Dashboard from './Dashboard';
import Projects from './Projects';
import About from './about';
import Users from './Users';
import Register from './Register';
import ChatRoom from './ChatRoom';
<%_ } _%>

const routes = [
    {
        path: '/',
        component: Dashboard,
        exact: true,
        key: 'dashboard'
    },
    {
        path: '/register',
        component: Register,
        key: 'register'
    },
    {
        path: '/login',
        component: Login,
        key: 'login'
    },
    {
        path: '/about',
        component: About,
        key: 'about'
    },
    {
        path: '/topics',
        component: Topics,
        key: 'topics'
    },
    {
        path: '/projects',
        component: Projects,
        key: 'projects'
    },
    {
        path: '/users',
        component: Users,
        key: 'users'
    },
    {
        path: '/chat',
        component: ChatRoom,
        key: 'chat'
    }
];

export default routes;
