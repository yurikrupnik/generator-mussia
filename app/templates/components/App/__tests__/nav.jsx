import React from 'react';
import MainNav from '../nav';
import routes from '../../routes';

jest.mock('../../routes'); // eslint-disable-line no-undef

const {
    test,
    expect,
    shallow
} = global;

test('renders <Nav /> component', () => {
    const wrapper = shallow(<MainNav routes={routes} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').length).toBe(2);
});
