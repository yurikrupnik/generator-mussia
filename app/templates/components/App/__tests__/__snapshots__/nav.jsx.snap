// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Nav /> component 1`] = `
<header>
  <button
    onClick={[Function]}
    type="button"
  >
    logout
  </button>
  <div
    key="hello"
  >
    <Link
      replace={false}
      to="/hello"
    >
      hello
    </Link>
  </div>
  <div
    key="world"
  >
    <Link
      replace={false}
      to="/world"
    >
      world
    </Link>
  </div>
</header>
`;
